pkg/estimator/hll/hll.go

```diff
@@ -241,6 +242,10 @@ func (h *Plus) MarshalBinary() (data []byte, err error) {
 		return nil, nil
 	}
 
+	if h.sparse {
+		h.mergeSparse()
+	}
```

Comment:
👍  thanks! This makes sense now. this was added because the `sparseList` has repeatable marshalling, but the `tmp_set` marshals in randomized order (it iterates a golang map). We maintain the `sparseList` as a compressed list of keys in sorted order.  So a given HLL set used to have many possible binary representations but now it always has exactly one representation.

The `tmp_set` is really just a runtime optimization to batch updates into the `sparseList`.